### DISCUSSION TITLE
Scale only argocd-server deployment

### DISCUSSION THREAD

**Discussion by danilorsilva** [argoproj/argo-cd#6137] [Q&A] [answered]
Hello all, i have a not HA argocd installation on my cluster and i want to increase the replica number from 1 to 3 for just argocd-server deployment.

Is it possible or i wil face some kind of problem?

thanks!

**Comment by jannfis** [comment, marked as answer]
Hi @danilorsilva. 

Since the `argocd-server` workload is mainly stateless (it keeps just a little state information in the Redis cache), you should face no problems increasing its replicas. The same is true for `argocd-repo-server` workload, which can also be scaled up independently.

The HA manifests also add a high available Redis installation, which the non-HA manifests don't bring. In general, you could also _upgrade_ your non-HA manifests with the HA-manifests, which will also add some anti-affinity rules to the workloads. However, it will require a cluster with a minimum of 3 nodes due to that.

HTH.

**Reply by danilorsilva** [reply]
Thank you so much @jannfis for your complete and detailed answer.

**Reply by emirot** [reply]
@jannfis I have a follow up question regarding application-controller as it's a statefulset.

From the doc https://argo-cd.readthedocs.io/en/stable/operator-manual/high_availability/

I m a bit confused if that one can run in HA with multiple replicas. Does it depends on the sharding algorithm used? 

Can I have two application-controller replica and only manage one cluster ?
Would that improve resiliency if a pod goes down.
 
 The official chart example in the chart is only 1: https://github.com/argoproj/argo-helm/tree/main/charts/argo-cd#ha-mode-with-autoscaling.

**Reply by emirot** [reply]
Got an answer there https://github.com/argoproj/argo-helm/discussions/2368#discussioncomment-7768412, well is that still accurate https://github.com/argoproj/argo-cd/blob/master/docs/proposals/rebalancing-clusters-across-shards-dynamically.md
